server/index.js

```diff
@@ -167,6 +165,36 @@ app.post('/api/cart', (req, res, next) => {
 
 });
 
+app.post('/api/orders', (req, res, next) => {
+  if (!req.session.cartId) {
+    return next(new ClientError('no current cart session', 400));
+  }
+
+  if (req.body.name && req.body.creditCard && req.body.shippingAddress) {
+    const sql = `
+      insert into "orders" ("cartId", "name", "creditCard", "shippingAddress")
+        values ($1, $2, $3, $4)
+        returning "createdAt",
+                  "creditCard",
+                  "name",
+                  "orderId",
+                  "shippingAddress";
+    `;
+
+    const values = [req.session.cartId, req.body.name, req.body.creditCard, req.body.shippingAddress];
+
+    db.query(sql, values)
+      .then(result => {
+        if (result.rows[0]) {
+          delete req.session.cartId;
+        }
```

Comment:
If the insert is successful at all and it includes a `returning` clause, then `result.rows[0]` is guaranteed to be truthy.